packages/manager/modules/nutanix/src/dashboard/general-info/redeploy/controller.js

```diff
@@ -0,0 +1,132 @@
+import {
+  REDEPLOY_INITIAL_CONFIG,
+  REDEPLOY_CUSTOM_CONFIG,
+  PRISM_CENTRAL_TYPES,
+  IPV4_REGEX,
+  IPV4_BLOCK_REGEX,
+  TRACKING_PREFIX,
+} from './constants';
+
+export default class NutanixGeneralInfoRedeployCtrl {
+  /* @ngInject */
+  constructor($translate, atInternet, Validator) {
+    this.$translate = $translate;
+    this.atInternet = atInternet;
+    this.Validator = Validator;
+    this.REDEPLOY_INITIAL_CONFIG = REDEPLOY_INITIAL_CONFIG;
+    this.REDEPLOY_CUSTOM_CONFIG = REDEPLOY_CUSTOM_CONFIG;
+    this.prismCentralTypes = PRISM_CENTRAL_TYPES;
+    this.IPV4_REGEX = IPV4_REGEX;
+    this.IPV4_BLOCK_REGEX = IPV4_BLOCK_REGEX;
+  }
+
+  $onInit() {
+    this.initalConfig = angular.copy(this.cluster.targetSpec);
+    this.displayPrismCentralIps = this.initalConfig.prismCentral.ips.join(',');
+    this.redeployMethod = REDEPLOY_INITIAL_CONFIG;
+    this.redeployMethods = {
+      initialConfig: [REDEPLOY_INITIAL_CONFIG],
+      customConfig: [REDEPLOY_CUSTOM_CONFIG],
+    };
+    this.initReplicationFactors();
+  }
+
+  initReplicationFactors() {
+    this.replicationFactors = {};
+    if (this.cluster.allowedRedundancyFactor.find((element) => element === 2)) {
```

Comment:
This need to be dynamic. This will break if 5 is introduced. Calculate the replication factors dynamically. Dont assume there will be only 2 and 3.